controller/signUpController.js

```diff
@@ -0,0 +1,110 @@
+const User = require('../models/user')
+const jwt = require('jsonwebtoken')
+const bcrypt = require('bcrypt')
+const secretKey = require('../config/secretKey')
+const dotenv = require('dotenv')
+const http = require('http-status-codes')
+const email_verify = require('./emailVerify')
+const statusMsg = require('../config/statusMsg')
+const SALTING = 10
+const { validationResult } = require('express-validator/check')
+const tokenGenerator = require('./authTokenGenerator')
+const createUser = require('../dbQuery/createUser')
+dotenv.config({
+    path: './config/.env'
+})
+
+exports.create = async (req, res, next) => {
+    const errors = validationResult(req)
+    if (!errors.isEmpty()) {
+        return res.status(http.UNPROCESSABLE_ENTITY).json({
+            "success": statusMsg.fail.msg,
+            "payload": "",
+            "error": {
+                "code": http.UNPROCESSABLE_ENTITY,
+                "message": errors.array()
+            }
+        })
+    }
+    try {
+        let role = null
+        if (req.body.user_role == 1) { role = "Mentor" }
+        else if (req.body.user_role == 0) { role = "Student" }
+        let hash = await bcrypt.hash(req.body.password, SALTING)
+        let email = await User.findOne({ email: req.body.email })
+        if (email) {
+            if (typeof email.password === 'undefined') {
+                email.password = hash
+                email.user_role = role
+                await email.save()
+                const response = {
+                    "data": email
+                }
+                res.status(http.CREATED).json({
+                    "success": statusMsg.success.msg,
+                    "payload": response
+                })
+            }
+        }
+        else if (!email) {
+            createUser.createUser(req, res)
+        }
+
+    }
+    catch (err) {
+        console.log(err)
+        err.status = http.CONFLICT
+        next(err)
+    }
+}
+
+exports.confirmation = async (req, res, next) => {
+    try {
+        if (req.query.token) {
+            decoded = await jwt.verify(req.query.token, secretKey.token.key)
```

Comment:
@AwaleRohin any reason on sending query param?

Reply:
because jwt token is on confirmation link query param so to get that jwt token